saleor/static/dashboard-next/components/TableHead/TableHead.tsx

```diff
@@ -1,38 +1,123 @@
-import { createStyles, withStyles, WithStyles } from "@material-ui/core/styles";
+import {
+  createStyles,
+  Theme,
+  withStyles,
+  WithStyles
+} from "@material-ui/core/styles";
+import { fade } from "@material-ui/core/styles/colorManipulator";
+import TableCell from "@material-ui/core/TableCell";
 import MuiTableHead, {
   TableHeadProps as MuiTableHeadProps
 } from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import Typography from "@material-ui/core/Typography";
+import * as classNames from "classnames";
 import * as React from "react";
 
-import TableActions from "../TableActions";
+import { Node } from "../../types";
+
+import i18n from "../../i18n";
+import Checkbox from "../Checkbox";
 
 export interface TableHeadProps extends MuiTableHeadProps {
+  disabled: boolean;
   selected: number;
+  items: Node[];
   toolbar: React.ReactNode | React.ReactNodeArray;
+  toggleAll: (items: Node[], selected: number) => void;
 }
 
-const styles = createStyles({
-  shrink: {}
-});
+const styles = (theme: Theme) =>
+  createStyles({
+    cell: {
+      padding: 0
+    },
+    checkboxSelected: {
+      backgroundColor: fade(theme.palette.primary.main, 0.05)
+    },
+    container: {
+      alignItems: "center",
+      display: "flex",
+      height: 47,
+      marginRight: -theme.spacing.unit * 2
+    },
+    padding: {
+      "&:last-child": {
+        padding: 0
+      }
+    },
+    root: {
+      backgroundColor: fade(theme.palette.primary.main, 0.05),
+      borderBottom: "1px solid rgba(224, 224, 224, 1)",
+      paddingLeft: 0,
+      paddingRight: 24
+    },
+    spacer: {
+      flex: 1
+    },
+    toolbar: {
+      "& > *": {
+        marginLeft: theme.spacing.unit
+      }
+    }
+  });
 
 const TableHead = withStyles(styles, {
   name: "TableHead"
 })(
   ({
-    children,
     classes,
+    children,
+    disabled,
+    items,
     selected,
+    toggleAll,
     toolbar,
     ...muiTableHeadProps
-  }: TableHeadProps & WithStyles<typeof styles>) => (
-    <MuiTableHead {...muiTableHeadProps}>
-      {selected ? (
-        <TableActions selected={selected}>{toolbar}</TableActions>
-      ) : (
-        children
-      )}
-    </MuiTableHead>
-  )
+  }: TableHeadProps & WithStyles<typeof styles>) => {
+    const [isSelected, setSelected] = React.useState(false);
+    return (
+      <MuiTableHead {...muiTableHeadProps}>
+        <TableRow>
+          <TableCell
+            padding="checkbox"
+            className={classNames({
+              [classes.checkboxSelected]: selected
+            })}
+          >
+            <Checkbox
+              checked={isSelected}
```

Comment:
Seems that master checkbox `checked` status should be `selected.length === items.length` instead of local state.

Reply:
Aww damn, I missed this one